gammapy/maps/_sparse.pyx

```diff
@@ -0,0 +1,207 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import, division, print_function, unicode_literals
+import numpy as np
+cimport numpy as np
+cimport cython
+
+
+@cython.cdivision(True)
+@cython.boundscheck(False)
+def binary_search(np.ndarray[np.int64_t, ndim=1] arr_in, np.int64_t idx,
+                  np.int64_t ilo, np.int64_t ihi):
+    """Perform a binary search for an element ``idx`` in a sorted integer
+    array ``arr_in``.  The search is restricted to the range between
+    ``ilo`` and ``ihi``.  This will return the index of the matching
+    element or the index at which ``idx`` would be inserted to
+    maintain sorted order.
+
+    Parameters
+    ----------
+    arr_in : `~numpy.ndarray`
+        Sorted array of integers.
+
+    idx : int
+        Value to be searched for.
+
+    """
+    
+    cdef int mid = 0
+    cdef int midval = 0
+
+    while(ilo < ihi):
+        mid = (ilo + ihi) // 2
+        midval = arr_in[mid]
+
+        if midval < idx:
+            ilo = mid + 1
+        elif midval > idx:
+            ihi = mid
+        else:
+            return mid
+
+    return ilo
+
+
+@cython.cdivision(True)
+@cython.boundscheck(False)
+def binary_search2(np.ndarray[np.int64_t, ndim=1] arr_in, np.int64_t idx,
```

Comment:
Remove `binary_search2`? It doesn't seem to be called anywhere.
Alternatively, please add a short comment saying what it is (e.g. "this is an alternative implementation of `binary_search2` that I want to keep around for performance testing").

Reply:
Removed the method.